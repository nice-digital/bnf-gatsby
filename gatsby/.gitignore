node_modules/
.cache/
public

# dotenv environment variables file
.env
.env.*
