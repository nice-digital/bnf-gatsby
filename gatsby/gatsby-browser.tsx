import {
	ShouldUpdateScrollArgs,
	type RouteUpdateArgs,
	type WrapPageElementBrowserArgs,
} from "gatsby";
import { type ReactElement } from "react";

import { Layout } from "@/components/Layout/Layout";

// Gatsby hook for when the route has changed on the client side
// See https://www.gatsbyjs.org/docs/browser-apis/#onRouteUpdate
export const onRouteUpdate = ({
	prevLocation,
	location,
}: RouteUpdateArgs): void => {
	if (prevLocation) {
		// Cast is needed because of https://github.com/gatsbyjs/gatsby/issues/29124
		const prevPath = (prevLocation as Location).pathname,
			path = location.pathname;

		// Push our own event to the dataLayer on page change rather than using the
		// 'gatsby-route-change' event built into gatsby-plugin-google-tagmanager.
		// Because gatsby-route-change is pushed on initial page load as well as route change,
		// AND because it doesn't use requestAnimationFrame to delay until the page title has been updated

		const sendPageView = () => {
			window.dataLayer.push({
				location: location.href,
				referrer: prevLocation.href,
			});
			// Don't consider hash changes to be a page view - pageviews only happy when the path changes
			if (prevPath != path) window.dataLayer.push({ event: "pageview" });
		};

		// Delay before push to the data layer, to make sure the page title has been updated
		// See https://github.com/gatsbyjs/gatsby/pull/10917/files#diff-bf0d94c8bf47d5c1687e342c2dba1e00R12-R13
		if ("requestAnimationFrame" in window) {
			window.requestAnimationFrame(() => {
				window.requestAnimationFrame(sendPageView);
			});
		} else {
			// simulate 2 rAF calls
			setTimeout(sendPageView, 32);
		}
	} else {
		window.dataLayer.push({
			location: location.href,
			referrer: document.referrer,
		});
	}
	// Default return statement when prevLocation is falsy
	return;
};

export const wrapPageElement = ({
	element,
	props,
}: WrapPageElementBrowserArgs): ReactElement => (
	<Layout {...props}>{element}</Layout>
);

/**
 * Gatsby hook for overriding scroll position
 * See https://www.gatsbyjs.org/docs/browser-apis/#shouldUpdateScroll
 */
export const shouldUpdateScroll = ({
	routerProps: { location },
	prevRouterProps,
	getSavedScrollPosition,
}: ShouldUpdateScrollArgs): boolean | string | [number, number] => {
	if (
		// If there's no previous route props we're coming from an external site, which means
		// we want to scroll to the hash (if there is one), and _not_ a saved scroll position
		!prevRouterProps ||
		// Or we're linking to as hash within the same page
		(prevRouterProps.location.pathname === location.pathname && location.hash)
	) {
		// Provide our own scroll to hash to avoid Gatsby using a stored scroll position
		requestAnimationFrame(() => {
			requestAnimationFrame(() => {
				const targetElement = location.hash
					? document.querySelector(location.hash)
					: null;

				if (targetElement) {
					targetElement.setAttribute("tabIndex", "-1");
					(targetElement as HTMLElement).focus();
					targetElement.scrollIntoView();
				}
			});
		});
		return false;
	}

	const savedScrollY = (getSavedScrollPosition(location)?.[1] || 0) as number;
	if (savedScrollY > 0) {
		window.scrollTo(0, savedScrollY);
		return false;
	}

	if (location.hash) {
		requestAnimationFrame(() => {
			requestAnimationFrame(() => {
				const targetElement = document.querySelector(location.hash);

				// Default to Gatsby's default behaviour if the element doesn't exist
				if (!targetElement) return true;

				// Provide our own scroll to hash to avoid Gatsby using a stored scroll position
				targetElement.setAttribute("tabIndex", "-1");
				(targetElement as HTMLElement).focus();
				targetElement.scrollIntoView();
				return false;
			});
		});
		return false;
	}

	requestAnimationFrame(() => {
		requestAnimationFrame(() => {
			const contentStartElement = document.getElementById("content-start");

			if (!contentStartElement) return true;

			contentStartElement.setAttribute("tabIndex", "-1");

			//scrolling to 0,0 to preserve existing BNF navigation behaviour
			window.scrollTo(0, 0);
			return false;
		});
	});
	// Default to scrolling to the content start element as the standard navigation behaviour

	return true;
};
