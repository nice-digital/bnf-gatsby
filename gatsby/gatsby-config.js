require("dotenv").config({
	path: `.env.${process.env.NODE_ENV}`,
});

const isBNF = process.env.GATSBY_SITE === "bnf";

if (!process.env.GATSBY_SEARCH_URL)
	throw new Error(
		"Env var GATSBY_SEARCH_URL isn't set. Did you forget to add it?"
	);

module.exports = {
	jsxRuntime: "automatic",
	siteMetadata: {
		isBNF,
		siteUrl: `https://${isBNF ? "bnf" : "bnfc"}.nice.org.uk`,
		siteTitleShort: isBNF ? "BNF" : "BNFC",
		siteTitleLong:
			"British National Formulary" + (isBNF ? "" : " for Children"),
	},
	plugins: [
		{
			resolve: `gatsby-source-bnf`,
			options: {
				feedURL: process.env.FEED_URL,
				userKey: process.env.FEED_USER_KEY,
				site: isBNF ? "bnf" : "bnfc",
			},
		},
		// Avoid errors like "ModuleNotFoundError: Module not found: Error: Can't resolve '@/components/Layout/Layout'" when using custom paths in tsconfig.json
		`gatsby-plugin-tsconfig-paths`,
		{
			resolve: `gatsby-plugin-sass`,
			options: {
				// Make CSS modules play nicely with TypeScript, see also global.d.ts for *.scss TS def
				cssLoaderOptions: {
					esModule: false,
					modules: {
						namedExport: false,
					},
				},
			},
		},
		"gatsby-plugin-catch-links",
		"gatsby-plugin-react-helmet",
		"gatsby-plugin-remove-generator",
		{
			resolve: `gatsby-plugin-sitemap`,
			options: {
				excludes: [`/search/`],
			},
		},
		{
			resolve: "gatsby-plugin-google-tagmanager",
			options: {
				id: "GTM-5H5L9GK",
				includeInDevelopment: true,
			},
		},
		{
			resolve: "gatsby-plugin-manifest",
			options: {
				icon: "src/images/icon.png",
			},
		},
	],
};
