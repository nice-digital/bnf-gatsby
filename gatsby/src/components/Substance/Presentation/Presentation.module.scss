@use '@nice-digital/nds-core/scss/core';

.packDefinitionListItem {
  margin-bottom: core.rem(core.$nds-spacing-small);

  &:last-child {
    margin-bottom: 0;
  }

  @include core.mq($from: md) {
    display: flex;
  }

  dt {
    font-weight: bold;

    @include core.mq($from: md) {
      flex: 1 1 30%;
    }
  }

  dd {
    margin: 0;

    > :first-child {
      margin-top: 0;
    }

    > :last-child {
      margin-bottom: 0;
    }

    @include core.mq($from: md) {
      flex: 1 1 70%;
    }
  }
}
