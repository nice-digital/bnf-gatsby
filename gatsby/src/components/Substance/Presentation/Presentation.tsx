import { FC } from "react";

import { FeedBorderlineSubstancePresentation } from "@nice-digital/gatsby-source-bnf";

import { Prep } from "@/components/Prep/Prep";
import { type QueryResult } from "@/utils";

import styles from "./Presentation.module.scss";

const nilValue = "Nil";

const GramValue = (props: {
	label: string;
	valueInGrams: string;
	constituents: string[];
}) => (
	<div className={styles.packDefinitionListItem}>
		<dt>{props.label}</dt>
		<dd>
			{props.valueInGrams === nilValue ? (
				props.valueInGrams
			) : (
				<>
					{props.valueInGrams}&nbsp;g{" "}
					{props.constituents.length > 0 ? props.constituents.join(", ") : null}
				</>
			)}
		</dd>
	</div>
);

export type PresentationProps = {
	presentation: QueryResult<FeedBorderlineSubstancePresentation>;
};

const Presentation: FC<PresentationProps> = ({
	presentation: {
		acbs,
		carbohydrateConstituents,
		carbohydrateGrams,
		energyKCal,
		energyKj,
		fatConstituents,
		fatGrams,
		fibreConstituents,
		fibreGrams,
		formulation,
		proteinConstituents,
		proteinGrams,
		specialCharacteristics,
		borderlineSubstancePreps,
		presentationNote,
	},
}) => {
	return (
		<>
			<dl>
				{formulation && (
					<div className={styles.packDefinitionListItem}>
						<dt>Formulation</dt>
						<dd>{formulation}</dd>
					</div>
				)}

				{energyKj || energyKCal ? (
					<div className={styles.packDefinitionListItem}>
						<dt>Energy</dt>
						<dd>
							{energyKj && <>{energyKj}&nbsp;kJ</>}
							{energyKj && energyKCal ? " / " : null}
							{energyKCal && <>{energyKCal}&nbsp;kcal</>}
						</dd>
					</div>
				) : null}

				{proteinGrams && (
					<GramValue
						label="Protein"
						valueInGrams={proteinGrams}
						constituents={proteinConstituents}
					/>
				)}

				{carbohydrateGrams && (
					<GramValue
						label="Carbohydrate"
						valueInGrams={carbohydrateGrams}
						constituents={carbohydrateConstituents}
					/>
				)}

				{fatGrams && (
					<GramValue
						label="Fat"
						valueInGrams={fatGrams}
						constituents={fatConstituents}
					/>
				)}

				{fibreGrams && (
					<GramValue
						label="Fibre"
						valueInGrams={fibreGrams}
						constituents={fibreConstituents}
					/>
				)}

				{specialCharacteristics.length > 0 ? (
					<div className={styles.packDefinitionListItem}>
						<dt>Special characteristics</dt>
						<dd>{specialCharacteristics.join(", ")}</dd>
					</div>
				) : null}

				{acbs.length > 0 ? (
					<div className={styles.packDefinitionListItem}>
						<dt>ACBS indications</dt>
						<dd dangerouslySetInnerHTML={{ __html: acbs.join("") }} />
					</div>
				) : null}
			</dl>

			{presentationNote && <p>{presentationNote}</p>}

			{borderlineSubstancePreps.map((preparation) => (
				<Prep key={preparation.ampId} prep={preparation}></Prep>
			))}
		</>
	);
};

export default Presentation;
