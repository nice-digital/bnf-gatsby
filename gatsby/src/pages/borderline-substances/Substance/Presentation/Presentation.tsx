import { FC } from "react";

import { FeedBorderlineSubstancePresentation } from "@nice-digital/gatsby-source-bnf";

import styles from "./../../{BnfBorderlineSubstancesTaxonomy.slug}.module.scss";
import Preparation from "./Preparation/Preparation";

export type PresentationProps = {
	presentation: FeedBorderlineSubstancePresentation;
};

const Presentation: FC<PresentationProps> = ({ presentation }) => {
	return (
		<>
			<dl>
				<div className={styles.packDefinitionListItem}>
					<dt>
						<strong>Formulation</strong>
					</dt>
					<dd>{presentation.formulation}</dd>
				</div>

				<div className={styles.packDefinitionListItem}>
					<dt>
						<strong>Energy (Kj)</strong>
					</dt>
					<dd>
						{presentation.energyKj}{" "}
						{presentation.energyKCal && `(${presentation.energyKCal})`}
					</dd>
				</div>

				<div className={styles.packDefinitionListItem}>
					<dt>
						<strong>Protein (g)</strong>
					</dt>
					<dd>
						{presentation.proteinGrams}
						{presentation.proteinConstituents &&
							`(${presentation.proteinConstituents})`}
					</dd>
				</div>

				<div className={styles.packDefinitionListItem}>
					<dt>
						<strong>Carbohydrate (g)</strong>
					</dt>
					<dd>
						{presentation.carbohydrateGrams}
						{presentation.carbohydrateConstituents &&
							`(${presentation.carbohydrateConstituents})`}
					</dd>
				</div>

				<div className={styles.packDefinitionListItem}>
					<dt>
						<strong>Fat (g)</strong>
					</dt>
					<dd>{presentation.fatGrams}</dd>
				</div>

				<div className={styles.packDefinitionListItem}>
					<dt>
						<strong>Fibre (g)</strong>
					</dt>
					<dd>{presentation.fibreGrams}</dd>
				</div>

				<div className={styles.packDefinitionListItem}>
					<dt>
						<strong>Special characteristics</strong>
					</dt>
					<dd>{presentation.specialCharacteristics}</dd>
				</div>

				<div className={styles.packDefinitionListItem}>
					<dt>
						<strong>Standard ACBS indications</strong>
					</dt>
					<dd>{presentation.acbs}</dd>
				</div>
			</dl>

			{presentation.borderlineSubstancePreps?.map((preparation, i) => (
				<Preparation key={i} preparation={preparation}></Preparation>
			))}
		</>
	);
};

export default Presentation;
