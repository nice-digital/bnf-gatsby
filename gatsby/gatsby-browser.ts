import { RouteUpdateArgs } from "gatsby";

// Gatsby hook for when the route has changed on the client side
// See https://www.gatsbyjs.org/docs/browser-apis/#onRouteUpdate
export const onRouteUpdate = ({
	prevLocation,
	location,
}: RouteUpdateArgs): void => {
	if (prevLocation) {
		// Cast is needed because of https://github.com/gatsbyjs/gatsby/issues/29124
		const prevPath = (prevLocation as Location).pathname,
			path = location.pathname;

		// Push our own event to the dataLayer on page change rather than using the
		// 'gatsby-route-change' event built into gatsby-plugin-google-tagmanager.
		// Because gatsby-route-change is pushed on initial page load as well as route change,
		// AND because it doesn't use requestAnimationFrame to delay until the page title has been updated

		const sendPageView = () => {
			window.dataLayer.push({
				location: location.href,
				referrer: prevLocation.href,
			});
			// Don't consider hash changes to be a page view - pageviews only happy when the path changes
			if (prevPath != path) window.dataLayer.push({ event: "pageview" });
		};

		// Delay before push to the data layer, to make sure the page title has been updated
		// See https://github.com/gatsbyjs/gatsby/pull/10917/files#diff-bf0d94c8bf47d5c1687e342c2dba1e00R12-R13
		if ("requestAnimationFrame" in window) {
			window.requestAnimationFrame(() => {
				window.requestAnimationFrame(sendPageView);
			});
		} else {
			// simulate 2 rAF calls
			setTimeout(sendPageView, 32);
		}
	} else {
		window.dataLayer.push({
			location: location.href,
			referrer: document.referrer,
		});
	}
};
