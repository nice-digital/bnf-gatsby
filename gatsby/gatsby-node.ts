import { CreateSchemaCustomizationArgs, CreateWebpackConfigArgs } from "gatsby";

import { initialFieldExtension } from "./src/field-extensions/initial";

/**
 * Gatsby hook for customizing the schema.
 * See https://www.gatsbyjs.org/docs/schema-customization/
 */
export const createSchemaCustomization = ({
	actions: { createFieldExtension, createTypes },
}: CreateSchemaCustomizationArgs): void => {
	createFieldExtension(initialFieldExtension);

	// Add initials for types used in A-Z lists, as it's easier for grouping
	// We don't add the initial field in the source plugin because grouping is a presentation concern
	const typeDefs = `
	  type BnfDrug implements Node {
		"The lowercase first letter of the title, used for grouping"
		initial: String! @initial(field: "title")
	  }
	  type BnfTreatmentSummary implements Node {
		"The lowercase first letter of the title, used for grouping"
		initial: String! @initial(field: "title")
	  }
	  type BnfInteractant implements Node {
		"The lowercase first letter of the title, used for grouping"
		initial: String! @initial(field: "title")
	  }
	`;
	createTypes(typeDefs);
};

/**
 * Gatsby hook for overriding webpack config
 * See https://www.gatsbyjs.com/docs/how-to/custom-configuration/add-custom-webpack-config/
 */
export const onCreateWebpackConfig = ({
	stage,
	actions,
	getConfig,
	plugins,
}: CreateWebpackConfigArgs): void => {
	// Disable order warnings from mini CSS plugin
	// See https://github.com/gatsbyjs/gatsby/discussions/30169#discussioncomment-621285
	if (stage === "build-javascript" || stage === "develop") {
		const config = getConfig();

		const miniCss = config.plugins.find(
			(plugin: () => void) => plugin.constructor.name === "MiniCssExtractPlugin"
		);

		if (miniCss) {
			miniCss.options.ignoreOrder = true;
		}

		actions.replaceWebpackConfig(config);

		actions.setWebpackConfig({
			plugins: [plugins.provide({ process: "process/browser" })],
		});
	}
};
