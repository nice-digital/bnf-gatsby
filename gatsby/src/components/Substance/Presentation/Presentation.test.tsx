import { render, waitFor, screen, within } from "@testing-library/react";
import { useStaticQuery } from "gatsby";

import Presentation, { PresentationProps } from "./Presentation";

const presentationProps: PresentationProps = {
	presentation: {
		acbs: [
			"<p><b>Standard ACBS indications:</b> Disease-related malnutrition, intractable malabsorption, pre-operative preparation of malnourished patients, dysphagia, proven inflammatory bowel disease, following total gastrectomy, short-bowel syndrome, bowel fistula except bowel fistula and pre-operative preparation of malnourished patients.</p>",
		],
		carbohydrateGrams: "13.0",
		energyKCal: "100",
		energyKj: "420",
		fatGrams: null,
		fibreConstituents: null,
		fibreGrams: "1.5",
		formulation: "Liquid (tube feed) per 100 mL",
		presentationNote: null,
		proteinGrams: "3.8",
		fatConstituents: [],
		carbohydrateConstituents: ["(sugars 0.9 g)"],
		proteinConstituents: ["milk, soya proteins"],
		borderlineSubstancePreps: [
			{
				manufacturer: "Fresenius Kabi Ltd",
				name: "Fresubin Original Fibre liquid",
				packs: [],
				ampId: "871011000001101",
				blackTriangle: false,
				controlledDrugSchedule: null,
				activeIngredients: [],
				sugarFree: null,
			},
			{
				manufacturer: "Fresenius Kabi Ltd",
				name: "Fresubin 1500 Complete liquid",
				packs: [],
				ampId: "16052411000001103",
				blackTriangle: false,
				controlledDrugSchedule: null,
				activeIngredients: [],
				sugarFree: null,
			},
		],
		rxAdvice: null,
		specialCharacteristics: [
			"Contains fish oil, residual lactose, soya. Gluten-free",
		],
	},
};

describe("presentation", () => {
	it("should match snapshot", () => {
		const { container } = render(<Presentation {...presentationProps} />);

		expect(container).toMatchSnapshot();
	});

	it("should render a definition list for each presentation, with a number of terms matching the data supplied", () => {
		render(<Presentation {...presentationProps} />);
		const allTerms = screen.getAllByRole("term");
		expect(allTerms.length).toBe(7);
	});

	it("should render each preparation", async () => {
		render(<Presentation {...presentationProps} />);
		const preparation = screen.getAllByRole("heading", { level: 3 });
		expect(preparation.length).toEqual(2);
		expect(preparation[0]).toHaveTextContent("Fresubin Original Fibre liquid");
	});
});
