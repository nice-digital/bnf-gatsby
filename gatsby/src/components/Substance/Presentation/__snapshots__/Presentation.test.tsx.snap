// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`presentation should match snapshot 1`] = `
<div>
  <dl>
    <div
      class="packDefinitionListItem"
    >
      <dt>
        Formulation
      </dt>
      <dd>
        Liquid (tube feed) per 100 mL
      </dd>
    </div>
    <div
      class="packDefinitionListItem"
    >
      <dt>
        Energy
      </dt>
      <dd>
        420
         kJ
         / 
        100
         kcal
      </dd>
    </div>
    <div
      class="packDefinitionListItem"
    >
      <dt>
        Protein
      </dt>
      <dd>
        3.8
         g
         
        milk, soya proteins
      </dd>
    </div>
    <div
      class="packDefinitionListItem"
    >
      <dt>
        Carbohydrate
      </dt>
      <dd>
        13.0
         g
         
        (sugars 0.9 g)
      </dd>
    </div>
    <div
      class="packDefinitionListItem"
    >
      <dt>
        Fibre
      </dt>
      <dd>
        1.5
         g
         
      </dd>
    </div>
    <div
      class="packDefinitionListItem"
    >
      <dt>
        Special characteristics
      </dt>
      <dd>
        Contains fish oil, residual lactose, soya. Gluten-free
      </dd>
    </div>
    <div
      class="packDefinitionListItem"
    >
      <dt>
        ACBS indications
      </dt>
      <dd>
        <p>
          <b>
            Standard ACBS indications:
          </b>
           Disease-related malnutrition, intractable malabsorption, pre-operative preparation of malnourished patients, dysphagia, proven inflammatory bowel disease, following total gastrectomy, short-bowel syndrome, bowel fistula except bowel fistula and pre-operative preparation of malnourished patients.
        </p>
      </dd>
    </div>
  </dl>
  <details
    class="details  "
  >
    <summary
      class="summary"
      data-tracking="Show"
    >
      <span
        class="toggleLabel label"
      >
        <svg
          aria-hidden="true"
          class="icon"
          fill="none"
          focusable="false"
          height="1em"
          viewBox="0 0 512 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M478.152 174c0 10.091-3.525 18.661-10.576 25.712l-186 186c-7.243 7.243-15.909 10.864-26 10.864-10.283 0-18.853-3.621-25.712-10.864l-186-186C36.621 192.853 33 184.283 33 174c0-10.091 3.621-18.757 10.864-26L65 126.576C72.424 119.525 81.09 116 91 116c10.091 0 18.661 3.525 25.712 10.576L255.576 265.44 394.44 126.576c7.051-7.051 15.621-10.576 25.712-10.576 9.909 0 18.576 3.525 26 10.576L467.576 148c7.051 7.424 10.576 16.091 10.576 26z"
            fill="currentColor"
          />
        </svg>
        Show
      </span>
       
      <h3
        class="prepHeading"
      >
        <span
          class="headingText"
        >
          Fresubin Original Fibre liquid
           
          <span
            class="manufacturer"
          >
            Fresenius Kabi Ltd
          </span>
        </span>
      </h3>
    </summary>
  </details>
  <details
    class="details  "
  >
    <summary
      class="summary"
      data-tracking="Show"
    >
      <span
        class="toggleLabel label"
      >
        <svg
          aria-hidden="true"
          class="icon"
          fill="none"
          focusable="false"
          height="1em"
          viewBox="0 0 512 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M478.152 174c0 10.091-3.525 18.661-10.576 25.712l-186 186c-7.243 7.243-15.909 10.864-26 10.864-10.283 0-18.853-3.621-25.712-10.864l-186-186C36.621 192.853 33 184.283 33 174c0-10.091 3.621-18.757 10.864-26L65 126.576C72.424 119.525 81.09 116 91 116c10.091 0 18.661 3.525 25.712 10.576L255.576 265.44 394.44 126.576c7.051-7.051 15.621-10.576 25.712-10.576 9.909 0 18.576 3.525 26 10.576L467.576 148c7.051 7.424 10.576 16.091 10.576 26z"
            fill="currentColor"
          />
        </svg>
        Show
      </span>
       
      <h3
        class="prepHeading"
      >
        <span
          class="headingText"
        >
          Fresubin 1500 Complete liquid
           
          <span
            class="manufacturer"
          >
            Fresenius Kabi Ltd
          </span>
        </span>
      </h3>
    </summary>
  </details>
</div>
`;
